lib/ProgramOptions/ProgramOptions.cpp

```diff
@@ -99,21 +99,47 @@ void ProgramOptions::printUsage() const {
 // hidden
 void ProgramOptions::printHelp(std::string const& search) const {
   bool const colors = (isatty(STDOUT_FILENO) != 0);
-  printUsage();
-
   TRI_TerminalSize ts = TRI_DefaultTerminalSize();
   size_t const tw = ts.columns;
   size_t const ow = optionsWidth();
 
+  std::string normalized = search;
+  if (normalized == "uncommon" || normalized == "hidden") {
+    normalized = ".";
+  }
+
+  bool showHidden = normalized == ".";
+
+  printUsage();
+
+  if (!showHidden) {
+    std::cout << "Common options (excluding hidden options) are:" << std::endl;
```

Comment:
Maybe this makes it a bit clearer?
```suggestion
    std::cout << "Common options (excluding hidden/uncommon options):" << std::endl;
```